pkg/apis/core/validation/shoot.go

```diff
@@ -1494,6 +1495,22 @@ func validateMaintenance(maintenance *core.Maintenance, fldPath *field.Path, wor
 	return allErrs
 }
 
+func validateCloudProfile(cloudProfileReference *core.CloudProfileReference, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if !features.DefaultFeatureGate.Enabled(features.UseNamespacedCloudProfile) || cloudProfileReference == nil {
+		return allErrs
+	}
```

Comment:
If we are setting the field to nil in the strategy when the feature gate is off, then do we need the first part of the check here?

Reply:
You are correct, logical mistake from my side. Addressed in https://github.com/gardener/gardener/pull/9440/commits/bc2bd226df5058cdde6c446ccabd89b024c97d49.